tests/ui/invalid_pymodule_args.stderr

```diff
@@ -1,4 +1,4 @@
-error: expected one of: `name`, `crate`, `module`, `submodule`
+error: expected one of: `name`, `crate`, `module`, `submodule`, `supports_free_threaded`
```

Comment:
bikeshed opportunity: should I call this `supports_free_threaded` or something else? supports_free_threading` is one more character but also feels less weird grammatically maybe?

I could also use GIL in the name since the relevant C API uses names like `GIL_NOT_USED` and `GIL_USED`.

Reply:
`supports_free_threading` seems slightly better to me; I think in the long run it will be `supports_free_threading = false` to opt-out, and the default won't need the argument, so longer but clearer seems the obvious choice to me.

Reply:
What about just `#[pymodule(free_threaded)]`?

Reply:
I guess to me that implies that it's _always_ free-threaded (which wouldn't be true on a GIL-enabled Python).

Reply:
Maybe `#[pymodule(requires_gil = false)]` ?

Reply:
How about `pymodule(gil_used)`, since the names of the `Py_mod_gil` slots that indicates an extension supports free-threading are `Py_MOD_GIL_NOT_USED` and `Py_MOD_GIL_USED`. With `gil_used` users are maybe somewhat more likely to link this with upstream documentation that is probably also useful for determining what they're supposed to do.

I think using `gil` in the name is fine since this is a low-level concern directly related to whether or not code is making strong assumptions about the GIL.

Reply:
In the latest version of this PR I renamed it to `gil_used = false`. In the long run that means users will do `pymodule(gil_used)` and otherwise not need to think about it.